### PR TITLE
Define _PyDict_Contains_KnownHash() FFI for CPython 3.10 or later

### PR DESCRIPTION
**Pull request by ijl**

### CODE REVIEW

File: pyo3-ffi/src/cpython/dictobject.rs [hunk 1]
```diff
@@ -63,6 +62,14 @@ extern "C" {
     // skipped _PyDictViewObject
     // skipped _PyDictView_New
     // skipped _PyDictView_Intersect
+
+    #[cfg(Py_3_10)]
+    pub fn _PyDict_Contains_KnownHash(
+        op: *mut PyObject,
+        key: *mut PyObject,
+        hash: Py_ssize_t,
```
Comment (messense) [inline pyo3-ffi/src/cpython/dictobject.rs:70]:
```suggestion
        hash: crate::Py_hash_t,
```

Maybe we should use [`Py_hash_t`](https://docs.rs/pyo3-ffi/latest/pyo3_ffi/type.Py_hash_t.html) for consistency?

**Review by messense — APPROVED** [review]
bors r+

**Comment by adamreichold** [comment]
bors retry

**Comment by adamreichold** [comment]
I think the failure of the careful CI job is due it using nightly Rust and some new warnings on `main` unrelated to the changes here. Will open a separate PR trying to fix this...

**Comment by adamreichold** [comment]
Retry after `main` was fixed.

bors retry

**Pull request merged**